src/interceptors/ClientRequest/NodeClientRequest.ts

```diff
@@ -37,6 +37,7 @@ export class NodeClientRequest extends ClientRequest {
     'ECONNREFUSED',
     'ECONNRESET',
     'EAI_AGAIN',
+    'ENETUNREACH',
```

Comment:
May I please ask you to add an integration test for this? You can grab your use case from the issue and put it `test/modules/http/regressions/http-socket-enetunreach.test.ts`.